_blog/deploying-laravel-to-kubernetes/index.md

```diff
@@ -0,0 +1,289 @@
+
+NAME                      DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+laravel-kubernetes-demo   1         1         1            1           57m
+```
+
+The output will be "1" for each. You want to have three available Pods so let's scale this up:
+
+```bash
+kubectl scale --replicas=3 deployment/laravel-kubernetes-demo
+deployment "laravel-kubernetes-demo" scaled
+```
+
+Done. You have replicated the first Pod to another two, giving you three Pods running this service. Running the `get deployment` command will verify this.
+
+```bash
+kubectl get deployment
+
+NAME                      DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+laravel-kubernetes-demo   3         3         3            3           59m
+```
+
+You can also see this in the Dashboard under Pods or in the Service detail screen.
+
+Now the application's deployment services are running on three 
+separate, replicated Pods.
+
+__Ingress__
+
+You've already achieved great things, you deployed the application and scaled the deployment. You have already seen the running application in the browser when pointed to the cluster's (Minikube) IP address and node's port number. Now, you will see how to access the application through an assigned URL as you would do when deploying to the cloud.
+
+Typically a Kubernetes cluster is firewalled from the internet, and the same is true for the Minikube cluster. To use a URL, in Kubernetes, you need Ingress. An Ingress is a set of rules to allow inbound connections to reach a Kubernetes cluster. The Ingress is necessary because, in Kubernetes, resources such as Pods only have IP addresses only routable by and within the cluster. Meaning that they are not accessible or reachable to and from the world outside.
+
+I have included an `ingress.yaml` file with the source code of this demo application with the following contents:
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: laravel-kubernetes-demo-ingress
+  annotations:
+    ingress.kubernetes.io/rewrite-target: /
+spec:
+  backend:
+    serviceName: default-http-server
+    servicePort: 80
+  rules:
+  - host: laravel-kubernetes.demo
+  - http:
+      paths:
+      - path: /
+        backend:
+          serviceName: laravel-kubernetes-demo
+          servicePort: 8181
+```
+Among the basic content you would expect from a Kubernetes resource file, this file defines a set of rules for the Kubernetes firewall to follow for inbound traffic. The `laravel-kubernetes.demo` URL will point to the Pod where the application is running, as previously labelled `laravel-kubernetes-demo` on port 8181.
+
+The Ingress resource is useless without an Ingress controller so you will need to create a new controller or use an existing one. This tutorial uses the Nginx Ingress controller for routing the traffic. Minikube (v0.14 and above) comes with the Nginx setup as an addon which you will need to enable manually:
+
+```bash
+minikube addons enable ingress
+```
```

Comment:
I suggest adding the following blockquote:

> Please note that it may take few minutes for Minikube to download and install Nginx as an ingress.

Reply:
👍

Reply:
Fair point! Let me explain what happens when you enable the add-on.

Under the bone, minikube is creating a deployment for Nginx. Same as the Laravel application. The Ingress is just another Pod with Nginx that acts as a reverse proxy. Nginx is packaged as a Docker container, exactly like your Laravel app.

So it may take a while to download the container and start the deployment.

In your case, I suspect the Docker image is cached locally. You can try and delete the cluster with:

```bash
$ minikube delete
```

and restart it with:

```bash
$ minikube start
```

If you enable the add-on you can quickly switch to another terminal and notice that something is happening with the pods:

```bash
$ kubectl get pods --all-namespaces
```

This part is very confusing, let me know if you want me to dive into more details.

Reply:
Yepp, I had the Nginx image cached. I verified this by deleting Minikube and reinstalling it.